src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/ClientCapabilitiesExtensions.cs

```diff
@@ -20,12 +20,20 @@ public static MarkupKind GetMarkupKind(this ClientCapabilities clientCapabilitie
     }
 
     public static bool SupportsMarkdown(this ClientCapabilities clientCapabilities)
-    {
-        return clientCapabilities.GetMarkupKind() == MarkupKind.Markdown;
-    }
+        => clientCapabilities.GetMarkupKind() == MarkupKind.Markdown;
 
     public static bool SupportsVisualStudioExtensions(this ClientCapabilities clientCapabilities)
-    {
-        return (clientCapabilities as VSInternalClientCapabilities)?.SupportsVisualStudioExtensions ?? false;
-    }
+        => clientCapabilities is VSInternalClientCapabilities { SupportsVisualStudioExtensions: true };
+
+    public static bool SupportsAnyCompletionListData(this ClientCapabilities clientCapabilities)
+        => clientCapabilities.SupportsCompletionListData() ||
+           clientCapabilities.SupportsCompletionListItemDefaultsData();
+
+    public static bool SupportsCompletionListData(this ClientCapabilities clientCapabilities)
+        => clientCapabilities.SupportsVisualStudioExtensions() &&
+           clientCapabilities.TextDocument?.Completion is VSInternalCompletionSetting { CompletionList.Data: true };
+
+    public static bool SupportsCompletionListItemDefaultsData(this ClientCapabilities clientCapabilities)
+        => clientCapabilities.TextDocument?.Completion?.CompletionListSetting?.ItemDefaults is { } defaults &&
+           Array.IndexOf(defaults, "data") >= 0;
```

Comment:
>Array.IndexOf(defaults, "data") >= 0; [](http://example.com/codeflow?start=11&length=37)

out of curiosity, why the switch from Contains to IndexOf?

Reply:
I assume it was to get away from using LINQ on an array.

Reply:
Yeah, the old code used Contains, but this file was using Array.IndexOf, so I just followed suit 🤷‍♂️